packages/strapi/lib/middlewares/public/index.js

```diff
@@ -97,12 +44,41 @@ module.exports = strapi => {
         })
```

Comment:
If you leave the defaultIndex option to true and create and index.html file in public it is not served right now. You need to set the `defer` option to false in the koaStatic middleware for this to work